src/overlays/actors/ovl_En_Tru/z_en_tru.c

```diff
@@ -25,91 +89,1189 @@ const ActorInit En_Tru_InitVars = {
     (ActorFunc)EnTru_Draw,
 };
 
-// static ColliderSphereInit sSphereInit = {
-static ColliderSphereInit D_80A8B2A0 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_SPHERE, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+#include "overlays/ovl_En_Tru/ovl_En_Tru.c"
+
+static Vec3f D_80A8B250 = { 0.0f, 0.02f, 0.0f };
+
+static Color_RGBA8 D_80A8B25C[] = {
+    { 255, 20, 60, 0 },   { 255, 235, 160, 0 }, { 20, 60, 255, 0 },
+    { 100, 100, 100, 0 }, { 255, 255, 200, 0 }, { 100, 100, 100, 0 },
+};
+
+static f32 D_80A8B274[] = { 60.0f, 255.0f, 60.0f };
+
+static UNK_TYPE D_80A8B280[] = {
+    0x0408F7E0, 0x0408F3E0, 0x0408EFE0, 0x0408EBE0, 0x0408E7E0, 0x0408E3E0, 0x0408DFE0, 0x0408DBE0,
+};
+
+static ColliderSphereInit sSphereInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_SPHERE,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 0, { { 0, 0, 0 }, 32 }, 100 },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_80A8B2CC = { 1, 20, 0, 0, MASS_IMMOVABLE };
+static CollisionCheckInfoInit2 sColChkInfoInit = { 1, 20, 0, 0, MASS_IMMOVABLE };
 
-#endif
+static ActorAnimationEntryS D_80A8B2D8[] = {
+    { &D_0600F9A0, 1.0f, 0, -1, 0, 0 },  { &D_0600F9A0, 1.0f, 0, -1, 0, -4 }, { &D_060108AC, 1.0f, 0, -1, 2, -4 },
+    { &D_06009348, 1.0f, 0, -1, 2, 0 },  { &D_0600EEDC, 1.0f, 0, -1, 0, -4 }, { &D_06015CA0, 1.0f, 0, -1, 0, 0 },
+    { &D_06015CA0, 1.0f, 0, -1, 0, -4 }, { &D_06014728, 1.0f, 0, -1, 2, 0 },  { &D_0601B5C4, 1.0f, 0, -1, 2, 0 },
+    { &D_06007FA0, 1.0f, 0, -1, 2, -4 }, { &D_06016B4C, 1.0f, 0, -1, 0, -4 }, { &D_06011F88, 1.0f, 0, -1, 2, -4 },
+    { &D_0600446C, 1.0f, 0, -1, 0, 0 },  { &D_06003698, 1.0f, 0, -1, 2, -4 }, { &D_06002BD8, 1.0f, 0, -1, 0, 0 },
+    { &D_0600446C, 1.0f, 0, -1, 0, 0 },
+};
+
+static Vec3f D_80A8B3D8 = { 0.0f, 24.0f, 16.0f };
+static Vec3f D_80A8B3E4 = { 0.0f, -3.0f, 3.0f };
+static Vec3f D_80A8B3F0 = { 0.0f, 0.5f, 0.0f };
+static Vec3f D_80A8B3FC = { 3000.0f, -800.0f, 0.0f };
+
+void func_80A85620(EnTruUnkStruct* arg0, Vec3f* arg1, f32 arg2, f32 arg3, f32 arg4) {
+    s16 i;
+
+    for (i = 0; i < 30; i++, arg0++) {
```

Comment:
```suggestion
    for (i = 0; i < ARRAY_COUNT(this->unk_394); i++, arg0++) {
```

Reply:
Can't do this, I don't have access to `this` for any of these functions. best would be a define.

Reply:
Oh good point. You can leave it as 30 then for now